### PR TITLE
add option for custom regex

### PR DESCRIPTION
**Pull request by rpolve**
As discussed in #79.

Users can now provide a custom regular expression to be used instead of default expressions.

```
  --regex REGEX, -E REGEX
                        Alternate custom regex to be used for all kinds of
                        matching. For example: --regex 'https?://.+\.\w+'
```

### CODE REVIEW

File: urlscan/urlscan.py [hunk 1]
```diff
@@ -412,15 +420,15 @@ def extracturls(mesg):
...
+    linechunks = [parse_text_urls(l, regex=regex) for l in lines]
 
     return extract_with_context(linechunks,
                                 lambda chunk: len(chunk) > 1 or
                                 (len(chunk) == 1 and chunk[0].url is not None),
                                 1, 1)
 
 
-def extracthtmlurls(mesg):
+def extracthtmlurls(mesg, regex=None):
```
Comment (firecat53) [inline urlscan/urlscan.py:431]: regex doesn't get used in this method. Should it be or can it be removed?

**Comment by firecat53** [comment]
Looks great! Thanks for your patience. Would you be willing to add a couple of more detailed example use cases in the README and manpage?

File: urlscan/urlscan.py (hunk 1 above)
Comment (rpolve) [reply urlscan/urlscan.py]: I apologize, I must have been operating on low caffeine. It is solved.

**Comment by rpolve** [comment]
> Looks great! Thanks for your patience. Would you be willing to add a couple of more detailed example use cases in the README and manpage?

It's done. Thank you for reviewing.

**Comment by firecat53** [comment]
Rebased (squashed) and merged. Thank you!

**Pull request closed without merging**